files/startup-script-stdlib.sh

```diff
@@ -0,0 +1,230 @@
+#! /bin/bash
+#
+# Copyright 2018 Google LLC. This software is provided as-is,
+# without warranty or representation for any use or purpose.
+# Your use of it is subject to your agreements with Google.
+
+# Standard library of functions useful for startup scripts.
+
+# Pending UX behaviors
+# TODO(jmccune): load_config_values to load user configuration
+# TODO(jmccune): call mandatory_argument() before E_MISSING_MANDATORY_ARG
+# TODO(jmccune): add -c <name> for alternate startup-script-custom name
+# Pending initialization functions:
+# TODO(jmccune): gsutil initialization w/ crcmod
+# Pending operational functions:
+# TODO(jmccune): get_from_bucket()
+# TODO(jmccune): setup_init_script()
+# TODO(jmccune): setup_sudoers()
+
+# These are outside init_global_vars so logging functions work with the most
+# basic case of `source startup-script-stdlib.sh`
+readonly SYSLOG_DEBUG_PRIORITY="${SYSLOG_DEBUG_PRIORITY:-syslog.debug}"
+readonly SYSLOG_INFO_PRIORITY="${SYSLOG_INFO_PRIORITY:-syslog.info}"
+readonly SYSLOG_ERROR_PRIORITY="${SYSLOG_ERROR_PRIORITY:-syslog.error}"
+# Global counter of how many times stdlib::init() has been called.
+STARTUP_SCRIPT_STDLIB_INITIALIZED=0
+
+# Error codes
+readonly E_RUN_OR_DIE=5
+readonly E_MISSING_MANDATORY_ARG=9
+readonly E_UNKNOWN_ARG=10
+
+stdlib::debug() {
```

Comment:
I'm not sure about the package name.  Previously (in the field) there was no package name, but I added `stdlib::` based on https://google.github.io/styleguide/shell.xml?showone=Function_Names#Function_Names

> Separate libraries with ::

Reply:
Sounds good, will keep stdlib for now.  Thanks Adrien.